lib/core/src/server/presets.js

```diff
@@ -0,0 +1,66 @@
+import path from 'path';
+import { logger } from '@storybook/node-logger';
+import serverRequire from './serverRequire';
+
+function loadPreset(preset) {
+  try {
+    if (typeof preset === 'string') {
+      return {
+        // eslint-disable-next-line global-require,import/no-dynamic-require
+        preset: require(preset),
+        options: {},
+      };
+    }
+
+    const { name, options } = preset;
+
+    return {
+      // eslint-disable-next-line global-require,import/no-dynamic-require
+      preset: require(name),
+      options,
+    };
+  } catch (e) {
+    logger.warn(`  Failed to load preset: ${JSON.stringify(preset)}`);
+    return false;
+  }
+}
+
+function loadPresets(presets) {
+  if (!presets || !Array.isArray(presets) || !presets.length) {
+    return [];
+  }
+
+  logger.info('Loading presets');
+
+  return presets.map(loadPreset).filter(preset => preset);
+}
+
+function applyPresets(presets, config, extension) {
+  if (!presets.length) {
+    return config;
+  }
+
+  return presets.reduce((accumulatedConfig, { preset, options }) => {
+    const extensionFn = preset[extension];
+
+    if (extensionFn && typeof extensionFn === 'function') {
+      return extensionFn.call(preset, accumulatedConfig, options);
+    }
+
+    return accumulatedConfig;
+  }, config);
+}
+
+function getPresets(configDir) {
+  const presets = serverRequire(path.resolve(configDir, 'presets'));
+  const loadedPresets = loadPresets(presets);
+
+  return {
+    extendBabel: config => applyPresets(loadedPresets, config, 'extendBabel'),
+    extendWebpack: config => applyPresets(loadedPresets, config, 'extendWebpack'),
```

Comment:
I will check this.

Reply:
This package looks interesting. Anyway, we can integrate it later. If we will go with it, it will be a main part of the core, so I would like to here other team members opinion as well =)